src/Modal/Modal.component.tsx

```diff
@@ -0,0 +1,271 @@
+// REACT
+import * as React from 'react';
+
+// ANCHOR
+import { Typography } from '..';
+import { Close } from '../Icon';
+import { IconSVGProps } from '../Icon/utils';
+import { sizes } from '../theme/sizes.theme';
+import { colors } from '../theme/colors.theme';
+
+// VENDOR
+import * as StyledReactModal from 'styled-react-modal';
+import styled, { css } from 'styled-components';
+import classnames from 'classnames';
+import { opacify } from 'polished';
+
+export type ModalSize = 'lg' | 'sm';
+
+const Sizes = {
+    lg: { width: 50, contentPadding: 2.5, footerHeight: 6.5 },
+    sm: { width: 25, contentPadding: 2, footerHeight: 5.5 },
+};
+const borderRadius = sizes.border.radius.modal;
+
+// Modal.Content
+// ------------------------------------------------------------------------------------------------------------------
+
+const StyledContent = styled.div<ModalContentProps>`
+    box-sizing: border-box;
+    width: 100%;
+
+    order: 0;
+    flex-grow: 1;
+`;
+
+interface ModalContentProps {
+    children?: any;
+    className?: string;
+}
+
+const ModalContent = ({ children, className }: ModalContentProps) => (
+    <StyledContent className={classnames('anchor-modal-content', className)}>
+        {children}
+    </StyledContent>
+);
+
+// Modal.Header
+// ------------------------------------------------------------------------------------------------------------------
+
+const StyledHeader = styled.div<ModalHeaderProps>`
+    box-sizing: border-box;
+    min-height: 4rem;
+    padding: 1.25rem 4.75rem 1.25rem 2rem;
+    width: 100%;
+    border-radius: ${borderRadius} ${borderRadius} 0 0;
+    margin-bottom: 1rem;
+
+    display: flex;
+    order: -1;
+
+    color: ${({ color = colors.charcoal.light }) => color};
+    background: ${({ background = colors.white.base }) => background};
+
+    // If the header exists, remove the Content's padding.
+    // We're preferring this selector to using Modal :first-child
+    // in case there is a close button placed before it, and using a
+    // double selector to ensure it takes precedence.
+    & ~ ${StyledContent}${StyledContent} {
+        padding-top: 0;
+    }
+`;
+
+interface ModalHeaderProps {
+    children?: any;
+    className?: string;
+    title?: string;
+    color?: string;
+    background?: string;
+}
+
+const ModalHeader = ({
+    children,
+    className,
+    title,
+    ...props
+}: ModalHeaderProps) => (
+    <StyledHeader
+        className={classnames('anchor-modal-header', className)}
+        {...props}
+    >
+        {title && (
+            <Typography scale={20} weight="bold">
+                {title}
+            </Typography>
+        )}
+        {children}
+    </StyledHeader>
+);
+
+// Modal.Footer
+// ------------------------------------------------------------------------------------------------------------------
+
+const StyledFooter = styled.div<ModalFooterProps>`
```

Comment:
We should isolate Header, Footer, & Content into separate files. I'm seeing a lot of conflation and I'd like to break some of these things up more granularly like Autocomplete, Menu, & List who have nested components.